include/revng/Support/Undirected.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/iterator.h"
+
+#include "revng/Support/Assert.h"
+
+namespace llvm {
+
+// TODO: Without the `llvm::concat` implementation limitation described below,
+//       we could offer a default implementation for
+//       `GraphTraits<Undirected<T>>` for any `T` for which you have a
+//       definition of both `GraphTraits<T>` and `GraphTraits<Inverse<T>>`
```

Comment:
I think we could still implement `GraphTraits<Undirected<T>>` using `UndirectedChildIterator`. The comment seems to imply this is not possible.

Reply:
Yeah, that comment was a bit confusing from the start, but I left there because now we have two orthogonals thing to improve the codebase, but that can be combined in improving the `Undirected` trait specialization.

The two things are:
1) Use the `GraphTraits` to provide the `UndirectedChildIterator` not only for `GenericGraph`, but using the traits. But this is explained in this TODO indeed https://github.com/revng/revng/blob/542a271c138b3255d8c126850db12d2ae3e761d5/include/revng/ADT/GenericGraph.h#L1001.
2) Implement our own `revng::concat` in place of `llvm::concat`, and this is commented with a TODO here https://github.com/revng/revng/blob/542a271c138b3255d8c126850db12d2ae3e761d5/include/revng/Support/Undirected.h#.

If 2) is done, we can drop the `UndirectedChildIterator` and re-implement the `GraphTraits<Undirected<T>>` using the `revng::concat`. Hence the comment above. But I also think that it was more confusing than anything else, so I dropped it, and the two `TODO`s combined should lead us to conclude what I said above.